### PR TITLE
Send accountsChanged notification for wallet_requestPermissions

### LINKED ISSUE #8740: accountsChanged is never emitted when wallet_requestPermissions is called directly
If a consumer directly calls `wallet_requestPermissions` and requests `eth_accounts`, `accountsChanged` will never be emitted to the corresponding conneciton.

### PR DESCRIPTION
**Pull request by rekmarks**
Fixes #8740

### CODE REVIEW

File: app/scripts/controllers/permissions/index.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ export class PermissionsController {
     this.getKeyringAccounts = getKeyringAccounts
     this._getUnlockPromise = getUnlockPromise
     this._notifyDomain = notifyDomain
-    this.notifyAllDomains = notifyAllDomains
+    this._notifyAllDomains = notifyAllDomains
```
Comment (rekmarks) [inline app/scripts/controllers/permissions/index.js:50]: This shouldn't be called directly by consumers.

File: app/scripts/controllers/permissions/index.js [hunk 2]
```diff
@@ -206,7 +208,7 @@ export class PermissionsController {
     const approval = this.pendingApprovals.get(id)
 
     if (!approval) {
-      log.error(`Permissions request with id '${id}' not found`)
+      log.debug(`Permissions request with id '${id}' not found`)
```
Comment (rekmarks) [inline app/scripts/controllers/permissions/index.js:211]: It's not actually a problem if this gets called multiple times, and we _do_ call `rejectPermissionsRequest` over-optimistically. No need for background errors.

File: app/scripts/controllers/permissions/index.js [hunk 3]
```diff
@@ -241,14 +243,15 @@ export class PermissionsController {
...
+   * Idempotent for a given id.
    *
    * @param {string} id - The id of the request rejected by the user
    */
   async rejectPermissionsRequest (id) {
     const approval = this.pendingApprovals.get(id)
 
     if (!approval) {
-      log.error(`Permissions request with id '${id}' not found`)
+      log.debug(`Permissions request with id '${id}' not found`)
```
Comment (rekmarks) [inline app/scripts/controllers/permissions/index.js:254]: See `approvePermissionsRequest`.

File: app/scripts/controllers/permissions/methodMiddleware.js [hunk 4]
```diff
@@ -81,10 +84,33 @@ export default function createMethodMiddleware ({
...
         break
     }
 
-    next()
+    // when this promise resolves, the response is on its way back
+    await next()
+
+    if (responseHandler) {
+      responseHandler()
+    }
```
Comment (rekmarks) [inline app/scripts/controllers/permissions/methodMiddleware.js:114]: This how to do response/return handlers for `createAsyncMiddleware` middlewares. Here, `next` does not take a return handler callback like in synchronous middlewares.

**Review by brad-decker — DISMISSED** [review]
LGTM

**Review by brad-decker — APPROVED** [review]

**Pull request merged**